app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -247,6 +247,14 @@ protected void startUpdateCheck() {
         uiController.downloadingUiState();
     }
 
+    private void initUpdateTaskProgressDisplay() {
+        if (CommCareApplication._().isConsumerApp()) {
+            showProgressDialog(DIALOG_CONSUMER_APP_UPGRADE);
```

Comment:
This seems a little sketch because you are trying to attach a dialog to a task that doesn't extend `CommCareTask`. Hence all the fragment management based off of `CommCareTask.taskId` _might_ not work.